tests/unit/main_html5/wrapper-api-tests.js

```diff
@@ -408,6 +439,208 @@ describe('main_html5 wrapper tests', function () {
+    setDvr(100, dvrWindowSize + 100);
+    $(element).triggerHandler("seeked");
+    // Current time should reflect shift calculated on wrapper.seeked() and shouldn't be updated after seeked event
+    $(element).triggerHandler("timeupdate");
+    params = vtc.notifyParameters[1];
+    expect(params.currentTime).to.be(dvrWindowSize / 2);
+  });
+
+  it('DVR: should adapt to changes in the DVR window', function() {
+    var params;
+    var dvrWindowSize = 1750;
+    var midpoint = dvrWindowSize / 2;
+    setDvr(0, dvrWindowSize);
+    element.currentTime = dvrWindowSize;
+    // Test seek with initial DVR window
+    wrapper.seek(midpoint);
+    $(element).triggerHandler("seeked");
+    $(element).triggerHandler("timeupdate");
+    params = vtc.notifyParameters[1];
+    expect(params.currentTime).to.be(midpoint);
+    expect(Number(params.currentLiveTime)).to.be(midpoint);
+    // Move DVR window forward
+    setDvr(100, dvrWindowSize + 100);
+    // Test seek with updated DVR window
+    wrapper.seek(midpoint);
+    $(element).triggerHandler("seeked");
+    $(element).triggerHandler("timeupdate");
+    params = vtc.notifyParameters[1];
+    expect(params.currentTime).to.be(midpoint);
+    expect(Number(params.currentLiveTime)).to.be(midpoint + 100);
+  });
+
+  it('DVR: should update time shift when resuming after a pause', function() {
+    var params;
+    var dvrWindowSize = 1800;
+    setDvr(0, dvrWindowSize);
+    element.currentTime = dvrWindowSize;
+    // Initial play
+    wrapper.play();
+    $(element).triggerHandler("playing");
+    // Check initial time shift
+    $(element).triggerHandler("timeupdate");
+    params = vtc.notifyParameters[1];
+    expect(params.currentTime).to.be(dvrWindowSize);
+    // Pause
+    wrapper.pause();
+    $(element).triggerHandler("pause");
+    // Move DVR window forward during pause
+    setDvr(100, dvrWindowSize + 100);
+    element.currentTime = dvrWindowSize;
+    // Resume playback
+    wrapper.play();
+    $(element).triggerHandler("playing");
+    $(element).triggerHandler("timeupdate");
+    params = vtc.notifyParameters[1];
+    // Time shift should reflect that we're falling behind the live playhead
+    expect(params.currentTime).to.be(dvrWindowSize - 100);
+  });
+
+  it('DVR: should NOT update time shift on initial play', function() {
+    var params;
+    var dvrWindowSize = 1800;
+    setDvr(0, dvrWindowSize);
+    element.currentTime = dvrWindowSize;
+    // Move DVR window forward before playing
+    setDvr(100, dvrWindowSize + 100);
+    element.currentTime = dvrWindowSize;
+    // Initial play
+    wrapper.play();
+    $(element).triggerHandler("playing");
+    // Time shift should not be affected by playing event
+    $(element).triggerHandler("timeupdate");
+    params = vtc.notifyParameters[1];
+    expect(params.currentTime).to.be(dvrWindowSize);
+  });
+
+  it('DVR: should NOT update time shift after PLAYING event if video was not paused', function() {
```

Comment:
Just curious, what are the circumstances we do not want to update the time shift, and why do we not want to update them?

Reply:
The main place where we update the time shift is when `seek()` is called, but then there are two special cases  where we also update:
- `seeked` event: In case the video was seeked with the native controls
- `playing` event: In order to show that you moved away from the live edge while the video was paused
For those two special cases we avoid updating the time shift when:
- The `seeked` event wasn't triggered by the native controls
- The `playing` event is fired for the first time. This is done so that we always start the video with the UI at the live edge, even if it's slightly offset.
- The `playing` event is fired but the video hadn't been paused. Safari fires `playing` after `seeked`, for example, so when this happens the time shift has already been updated and the playhead might jump a little if we update it again.

Reply:
Got it, thanks!